### ISSUE TITLE
Improve Error Logging in PathTraversal Plugin

### ISSUE THREAD

**Issue by davewichers** [zaproxy/zaproxy#2624]
I'm having a real hard time getting a scan with ZAP against OWASP Benchmark 1.2 to complete. I think the primary culprit is its getting stuck during the PathTraversal scans.  I looked in the log and see a few of the stack traces I list below. Can we improve these error messages in 2 ways:

1) Add the timeout info. I.e., how long did it wait before it gave up?
2) Add the URL and parameters sent in. Otherwise, its very difficult to figure out what ZAP was trying to do that causes this WARNING. I think it should show the GET with params, or the POST with the body params. Probably doesn't need to specify all the headers, but maybe it should, to be complete.
```
2016-06-25 16:35:34,555 WARN  TestPathTraversal - Error scanning parameters for Path Traversal: Read timed out
java.net.SocketTimeoutException: Read timed out
        at java.net.SocketInputStream.socketRead0(Native Method)
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
        at java.net.SocketInputStream.read(SocketInputStream.java:170)
        at java.net.SocketInputStream.read(SocketInputStream.java:141)
        at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
        at sun.security.ssl.InputRecord.read(InputRecord.java:503)
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
        at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930)
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
        at org.apache.commons.httpclient.HttpParser.readRawLine(HttpParser.java:78)
        at org.apache.commons.httpclient.HttpParser.readLine(HttpParser.java:106)
        at org.apache.commons.httpclient.HttpConnection.readLine(Unknown Source)
        at org.apache.commons.httpclient.MultiThreadedHttpConnectionManager$HttpConnectionAdapter.readLine(MultiThreadedHttpConnectionManager.java:1413)
        at org.apache.commons.httpclient.HttpMethodBase.readStatusLine(Unknown Source)
        at org.apache.commons.httpclient.HttpMethodBase.readResponse(Unknown Source)
        at org.apache.commons.httpclient.HttpMethodBase.execute(Unknown Source)
        at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(Unknown Source)
        at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(Unknown Source)
        at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
        at org.parosproxy.paros.network.HttpSender.executeMethod(Unknown Source)
        at org.parosproxy.paros.network.HttpSender.runMethod(Unknown Source)
        at org.parosproxy.paros.network.HttpSender.send(Unknown Source)
        at org.parosproxy.paros.network.HttpSender.sendAuthenticated(Unknown Source)
        at org.parosproxy.paros.network.HttpSender.sendAndReceive(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(Unknown Source)
        at org.zaproxy.zap.extension.ascanrules.TestPathTraversal.sendAndCheckPayload(TestPathTraversal.java:459)
        at org.zaproxy.zap.extension.ascanrules.TestPathTraversal.scan(TestPathTraversal.java:287)
        at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scanVariant(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(Unknown Source)
        at org.parosproxy.paros.core.scanner.AbstractPlugin.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by kingthorin** [comment]
Your probably get more useful logs if you enable debug: https://github.com/zaproxy/zaproxy/wiki/FAQlogging

<sub>Code ref: https://github.com/zaproxy/zap-extensions/blob/2e91b740e3275ea71dc51f0ae98f050fd36cca2b/src/org/zaproxy/zap/extension/ascanrules/TestPathTraversal.java#L437</sub>

**Comment by thc202** [comment]
1) The value of the time out should be the one defined in the options. [1]
2) Yeah, it would be better if it included more information of the attack by default.

[1] https://github.com/zaproxy/zap-core-help/wiki/HelpUiDialogsOptionsConnection#timeout-in-seconds

**Comment by davewichers** [comment]
So, are we going to open a ticket for these suggested improvements? I'm
happy to if you want me to.

-Dave

**Comment by thc202** [comment]
Released in version 24 of Active scanner rules add-on.

**Comment by thc202** [comment]
(For the record both changes were addressed in zaproxy/zap-extensions#451.)
